### PR TITLE
add countdown timer

### PR DESCRIPTION
**Pull request by wjrjerome**

### CODE REVIEW

File: common/countdown/countdown.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"time"
+
+	"github.com/XinFinOrg/XDPoSChain/log"
+)
+
+type CountdownTimer struct {
+	lock            sync.RWMutex // Protects the Initilised field
+	resetc          chan int
+	quitc           chan chan struct{}
+	Initilised      bool
```
Comment (wgr523) [inline common/countdown/countdown.go]: should `Initilised` be public?

**Review by wgr523 — APPROVED** [review]

File: common/countdown/countdown.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+// Completely stop the countdown timer from running.
+func (t *CountdownTimer) StopTimer() {
+	q := make(chan struct{})
+	t.quitc <- q
+	<-q
+}
+
+// Reset will start the countdown timer if it's already stopped, or simply reset the countdown time back to the defual `duration`
+func (t *CountdownTimer) Reset() {
+	if !t.getInitilisedValue() {
```
Comment (liam-lai) [inline common/countdown/countdown.go:39]: getInitilisedValue can rename like `haveInitilised`, return true mean yes, have already Initialised.

File: common/countdown/countdown_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+	// Now the countdown is paused after calling the callback function, let's reset it again
+	assert.True(t, countdown.getInitilisedValue())
+	expectedTimeAfterReset := time.Now().Add(5000 * time.Millisecond)
+	countdown.Reset()
+	<-called
+	// Always initilised
+	assert.True(t, countdown.getInitilisedValue())
+	if time.Now().After(expectedTimeAfterReset) {
+		fmt.Println("Correctly reset the countdown second time")
```
Comment (liam-lai) [inline common/countdown/countdown_test.go:68]: convert fmt. to `t.Info` or debug

File: common/countdown/countdown.go (hunk 2 above)
Comment (liam-lai) [reply common/countdown/countdown.go:39]: ok, I think `initilised` is even simpler.

File: common/countdown/countdown_test.go (hunk 3 above)
Comment (wjrjerome) [reply common/countdown/countdown_test.go:68]: t.Log

**Pull request merged**